azure/Guidelines.md

```diff
@@ -474,8 +472,23 @@ Overall, this is a very brittle design that leads to a poor developer experience
 ## Common API Patterns
 
 ### Performing an Action
+The REST specificaiton is used to model the state of a resource, and is primarily intended to handle CRUD (Create, Read, Update, Delete) operations. However, many services have the requirement to perform an action on a resource, e.g. getting the thumbnail of an image, sending an SMS message. To perform an action on a resource, we introduce the concept of a "controller resource," or simply "controller."
```

Comment:
I think the notion of a "controller resource" is confusing.  These are _not_ "resources" in the way we generally use this term in REST APIs -- you cannot "create" or "delete" or "get" them.  Why not simply call these what they are: "actions"?
> To perform an action on a resource, issue a request using the resource URL with a single additional path segment with the name of the action.

Reply:
+1. I like your simplified version *much* better and will include it. In this case, I think I pulled too much from the existing literature.